lib/internal/Magento/Framework/View/Helper/SecureHtmlRenderer.php

```diff
@@ -111,16 +111,21 @@ public function renderEventListenerAsTag(
             function {$listenerFunction} () {
                 {$attributeJavascript};
             }
-            var {$elementName} = document.querySelector("{$elementSelector}");
-            if ({$elementName}) {
-                {$elementName}.{$eventName} = function (event) {
-                    var targetElement = {$elementName};
-                    if (event && event.target) {
-                        targetElement = event.target;
+            var {$elementName}Array = document.querySelectorAll("{$elementSelector}");
+
+            {$elementName}Array.forEach(function(element){
```

Comment:
We need to make sure that `document.querySelectorAll` will return an empty node list if there's no `{$elementSelector}` on the page. @saphaljha have you tested this scenario?

Reply:
@rogyar , Thank you for suggestion, I have updated the code for validate empty nodeList. Also, covered MFTF.